ngraph/core/src/op/interpolate.cpp

```diff
@@ -233,11 +233,22 @@ void op::v4::Interpolate::validate_and_infer_types()
         return;
     }
 
+    const auto input_rank = input_shape.rank().get_length();
+
+    if (input_values().size() == 4 && !is_type<op::Constant>(input_value(3).get_node()))
```

Comment:
Please, add a comment here describing the logic and add a unit test for the `validate_and_infer_types` function for this case.

Reply:
Done.